src/components/json-tree.js

```diff
@@ -55,13 +55,30 @@ export default class JsonTree extends LitElement {
         border: 1px solid var(--border-color);
       }
 
+      /* Expand a block from 0 to something visible, then to probably full screen, then beyond most screens.
+         Unlike with a transition, the final CSS can have an unbounded height, which will take effect after the animation. */
+      @keyframes expand-height {
+        0% { max-height: 0; }
+        50% { max-height: 100px; }
+        95% { max-height: 1000px; }
+        100% { max-height: 5000px; }
+      }
+      /* Inverse of the above, collapsing quickly at first (to avoid a delay if the element is already quite short)
+         then slowing towards 0. */
+      @keyframes collapse-height {
```

Comment:
I feel like there was a "reverse" for key frames somewhere, does that not actually exist?

Reply:
There is an `animation-direction` property ... but it doesn't work for this use case. The problem is that if you switch between two classes, and the only difference is `animation-direction`, the animation doesn't restart - the browser just says "well, I've already played the animation for this element, so direction is irrelevant".

If the two classes have different `animation-name` properties (referencing different key frames), then the browser says "ooh, a new animation has been assigned to this element, I'll start it".

(This took me some time to discover and get my head around.)

Reply:
Hmm weird. I wonder if the `<details>` html tag solves this problem, haven't really looked into that one, but that triggered that though. No matter, I can understand that making sense.

Reply:
Hmm weird. I wonder if the `<details>` html tag solves this problem, haven't really looked into that one, but that triggered that though. No matter, I can understand that making sense.